wagtail_streamfield_migration_toolkit/test/tests/test_block_structure.py

```diff
@@ -0,0 +1,422 @@
+            content__2__nestedstruct__list1__value="a",
+            content__3="simplestruct",
+        ).content.raw_data
+        cls.raw_data = raw_data
+
+    @expectedFailure
+    def test_rename(self):
+        altered_raw_data = apply_changes_to_raw_data(
+            self.raw_data,
+            "nestedstruct.struct1.char1",
+            RenameBlockOperation(new_name="renamed1"),
+        )
+
+        self.assertNotIn("char1", altered_raw_data[1]["value"]["struct1"]["value"])
+        self.assertNotIn("char1", altered_raw_data[2]["value"]["struct1"]["value"])
+        self.assertIn("renamed1", altered_raw_data[2]["value"]["struct1"]["value"])
+        self.assertIn("renamed1", altered_raw_data[2]["value"]["struct1"]["value"])
+
+    def test_rename_rest_intact(self):
+        altered_raw_data = apply_changes_to_raw_data(
+            self.raw_data,
+            "nestedstruct.struct1.char1",
+            RenameBlockOperation(new_name="renamed1"),
+        )
+
+        self.assertEqual(len(altered_raw_data), 4)
+
+        self.assertEqual(altered_raw_data[0]["type"], "char1")
+        self.assertEqual(altered_raw_data[1]["type"], "nestedstruct")
+        self.assertEqual(altered_raw_data[2]["type"], "nestedstruct")
+        self.assertEqual(altered_raw_data[3]["type"], "simplestruct")
+
+        self.assertIn("char1", altered_raw_data[1]["value"])
+        self.assertIn("stream1", altered_raw_data[1]["value"])
+        self.assertIn("struct1", altered_raw_data[1]["value"])
+        self.assertIn("list1", altered_raw_data[1]["value"])
+
+        self.assertIn("char1", altered_raw_data[3]["value"])
+        self.assertIn("char2", altered_raw_data[3]["value"])
+
+        self.assertIn("char2", altered_raw_data[1]["value"]["struct1"])
+        self.assertIn("char2", altered_raw_data[2]["value"]["struct1"])
+
+    @expectedFailure
+    def test_remove(self):
+        altered_raw_data = apply_changes_to_raw_data(
+            self.raw_data, "nestedstruct.struct1.char1", RemoveBlockOperation()
+        )
+
+        self.assertEqual(len(altered_raw_data[1]["value"]["struct1"]), 1)
+        self.assertEqual(len(altered_raw_data[2]["value"]["struct1"]), 1)
+
+        self.assertNotIn("char1", altered_raw_data[1]["value"]["struct1"])
+        self.assertNotIn("char1", altered_raw_data[2]["value"]["struct1"])
+
+    def test_remove_rest_intact(self):
+        altered_raw_data = apply_changes_to_raw_data(
+            self.raw_data, "nestedstruct.struct1.char1", RemoveBlockOperation()
+        )
+
+        self.assertEqual(len(altered_raw_data), 4)
+        self.assertEqual(len(altered_raw_data[1]["value"]), 4)
+        self.assertEqual(len(altered_raw_data[2]["value"]), 4)
+        self.assertEqual(len(altered_raw_data[3]["value"]), 2)
+
+        self.assertIn("char2", altered_raw_data[1]["value"]["struct1"])
+        self.assertIn("char2", altered_raw_data[2]["value"]["struct1"])
+
+
+class FieldStreamStreamChildBlockTest(TestCase):
```

Comment:
I think it would probably be best to remove classes if they're completely empty - just put TODOs in for them